pkg/controller/podautoscaler/horizontal.go

```diff
@@ -182,45 +182,55 @@ func (a *HorizontalController) reconcileAutoscaler(hpa extensions.HorizontalPodA
 	cmStatus := ""
 	cmTimestamp := time.Time{}
 
-	if hpa.Spec.CPUUtilization != nil {
-		cpuDesiredReplicas, cpuCurrentUtilization, cpuTimestamp, err = a.computeReplicasForCPUUtilization(hpa, scale)
-		if err != nil {
-			a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeReplicas", err.Error())
-			return fmt.Errorf("failed to compute desired number of replicas based on CPU utilization for %s: %v", reference, err)
-		}
-	}
+	desiredReplicas := 0
+	timestamp := time.Now()
 
-	if cmAnnotation, cmAnnotationFound := hpa.Annotations[HpaCustomMetricsTargetAnnotationName]; cmAnnotationFound {
-		cmDesiredReplicas, cmStatus, cmTimestamp, err = a.computeReplicasForCustomMetrics(hpa, scale, cmAnnotation)
-		if err != nil {
-			a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeCMReplicas", err.Error())
-			return fmt.Errorf("failed to compute desired number of replicas based on Custom Metrics for %s: %v", reference, err)
+	if currentReplicas > hpa.Spec.MaxReplicas {
+		desiredReplicas = hpa.Spec.MaxReplicas
+	} else if hpa.Spec.MinReplicas != nil && currentReplicas < *hpa.Spec.MinReplicas {
+		desiredReplicas = *hpa.Spec.MinReplicas
+	} else if currentReplicas == 0 {
+		desiredReplicas = 1
+	} else {
+		// All basic scenarios covered, the state should be sane, lets use metrics.
+
+		if hpa.Spec.CPUUtilization != nil {
+			cpuDesiredReplicas, cpuCurrentUtilization, cpuTimestamp, err = a.computeReplicasForCPUUtilization(hpa, scale)
+			if err != nil {
+				a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeReplicas", err.Error())
+				return fmt.Errorf("failed to compute desired number of replicas based on CPU utilization for %s: %v", reference, err)
+			}
 		}
-	}
 
-	desiredReplicas := 0
-	timestamp := time.Time{}
+		if cmAnnotation, cmAnnotationFound := hpa.Annotations[HpaCustomMetricsTargetAnnotationName]; cmAnnotationFound {
+			cmDesiredReplicas, cmStatus, cmTimestamp, err = a.computeReplicasForCustomMetrics(hpa, scale, cmAnnotation)
+			if err != nil {
+				a.eventRecorder.Event(&hpa, api.EventTypeWarning, "FailedComputeCMReplicas", err.Error())
+				return fmt.Errorf("failed to compute desired number of replicas based on Custom Metrics for %s: %v", reference, err)
+			}
+		}
 
-	if cpuDesiredReplicas > desiredReplicas {
-		desiredReplicas = cpuDesiredReplicas
-		timestamp = cpuTimestamp
-	}
-	if cmDesiredReplicas > desiredReplicas {
-		desiredReplicas = cmDesiredReplicas
-		timestamp = cmTimestamp
-	}
+		if cpuDesiredReplicas > desiredReplicas {
```

Comment:
This looks like a bug in github. It shows wrong indentation :/ When I click "View" with whole files it's working as expected.